src/Functions/FunctionHelpers.cpp

```diff
@@ -298,4 +299,31 @@ bool isDecimalOrNullableDecimal(const DataTypePtr & type)
     return isDecimal(assert_cast<const DataTypeNullable *>(type.get())->getNestedType());
 }
 
+void checkFunctionArgumentSizes(const ColumnsWithTypeAndName & arguments [[maybe_unused]], size_t input_rows_count [[maybe_unused]])
+{
+    if (!arguments.empty())
+    {
+        /// Note that ideally this check should be simpler and we should check that all columns should either be const
+        /// or have exactly size input_rows_count
+        /// For historical reasons this is not the case, and many functions rely on the size of the first column
+        /// to decide which is the size of all the inputs
+        /// Hopefully this will be slowly improved in the future
+
+        if (!isColumnConst(*arguments[0].column))
```

Comment:
Maybe it is better to check that all non const columns? In this case we need to find first non_const column instead of this if-statement

Reply:
It makes sense. I'll need to check how we deal with const to decide to "materialize" it.